boot/bootstate20.go

```diff
@@ -150,9 +160,15 @@ func (u20 *bootStateUpdate20) commit() error {
 		}
 	}
 
-	// TODO:UC20: reseal against the TPM here with the updated modeenv written
-	//            but before doing any post-modeenv tasks so if we got rebooted
-	//            in between, we are still able to boot properly
+	if u20.dev != nil {
+		model := u20.dev.Model()
+		if model.Grade() != asserts.ModelGradeUnset {
+			// reseal if needed
+			if err := resealKeyToModeenv(model, u20.writeModeenv); err != nil {
```

Comment:
we'll have to think whether to use the fact we have written the modeenv in the if block before a hint to expect or not a reseal, maybe it's useful with unasserted kernel, anyway this can be a follow up